service/src/main/kotlin/fi/espoo/evaka/application/PendingDecisionEmailService.kt

```diff
@@ -0,0 +1,208 @@
+// SPDX-FileCopyrightText: 2017-2021 City of Espoo
+//
+// SPDX-License-Identifier: LGPL-2.1-or-later
+
+package fi.espoo.evaka.application
+
+import fi.espoo.evaka.daycare.domain.Language
+import fi.espoo.evaka.decision.getDecision
+import fi.espoo.evaka.emailclient.IEmailClient
+import fi.espoo.evaka.pis.getPersonById
+import fi.espoo.evaka.shared.async.AsyncJobRunner
+import fi.espoo.evaka.shared.async.SendPendingDecisionEmail
+import fi.espoo.evaka.shared.db.Database
+import mu.KotlinLogging
+import org.jdbi.v3.core.kotlin.mapTo
+import org.springframework.core.env.Environment
+import org.springframework.stereotype.Service
+import java.time.Instant
+import java.util.UUID
+
+private val logger = KotlinLogging.logger { }
+
+@Service
+class PendingDecisionEmailService(
+    private val asyncJobRunner: AsyncJobRunner,
+    private val emailClient: IEmailClient,
+    private val env: Environment
+) {
+    init {
+        asyncJobRunner.sendPendingDecisionEmail = ::doSendPendingDecisionsEmail
+    }
+
+    val fromAddress = env.getRequiredProperty("mail_reply_to_address")
+
+    fun doSendPendingDecisionsEmail(db: Database, msg: SendPendingDecisionEmail) {
+        logger.info("Sending pending decision reminder email to decision ${msg.decisionId}")
+        sendPendingDecisionEmail(db, msg.decisionId)
+    }
+
+    fun scheduleSendPendingDecisionsEmails(db: Database.Connection): Int {
+        val jobCount = db.transaction { tx ->
+            tx.createUpdate("DELETE FROM async_job WHERE type = 'SEND_PENDING_DECISION_EMAIL' AND (claimed_by IS NULL OR completed_at IS NOT NULL)")
+                .execute()
+
+            val pendingDecisions = tx.createQuery(
+                """
+SELECT DISTINCT(d.id)
+FROM decision d
+WHERE d.status = 'PENDING'
+AND d.resolved IS NULL
+AND d.sent_date < current_date - INTERVAL '1 week'
+AND d.pending_decision_emails_sent_count < 2
+AND d.pending_decision_email_sent IS NULL or d.pending_decision_email_sent < current_date - INTERVAL '1 week'
+"""
+            )
+                .mapTo<UUID>()
+                .list()
+
+            logger.info("PendingDecisionEmailService: Scheduling sending ${pendingDecisions.size} pending decision emails")
+
+            pendingDecisions.forEach { decisionId ->
+                asyncJobRunner.plan(
+                    tx,
+                    payloads = listOf(
+                        SendPendingDecisionEmail(
+                            decisionId = decisionId
+                        )
+                    ),
+                    runAt = Instant.now(),
+                    retryCount = 10
+                )
+            }
+
+            pendingDecisions.size
+        }
+
+        asyncJobRunner.scheduleImmediateRun()
+
+        return jobCount
+    }
+
+    fun sendPendingDecisionEmail(db: Database, decisionId: UUID) {
+        try {
+            db.read { tx ->
+                val decision = getDecision(tx.handle, decisionId) ?: run {
```

Comment:
If you just throw an exception `run` is unnecessary. You can just do `?: throw ...`.